src/runtime/algorithm.go

```diff
@@ -23,7 +23,12 @@ func fastrand() uint32 {
 	return xorshift32State
 }
 
-var xorshift32State uint32 = 1
+func init() {
+	xorshift64State = uint64(nanotime() | 1) // protect against nanotime() returning 0
```

Comment:
So this truncates 1 bit off `nanotime` for the seed. Would it make sense to get nanotime, then if 0, set to 1 or some NOMS 32-bit number?

Reply:
I guess an alternative would be to use `nanotime() | 1<<63`, or `nanotime() + 1` (I don't think `nanotime` goes negative).

Reply:
Unfortunately, `nanotime()` doesn't really add any randomness on baremetal because it starts at 0 on reset (so will still be 0 or very close to it in `init()`, in a very predictable way).

There are two ways this can be handled:

  * It could be left at a default value like 1 on baremetal (keeping the non-random behavior).
  * It could also use the `hardwareRand` function to obtain real random numbers from the hardware, if that's supported (it is on most 32-bit baremetal chips and on all common OSes). That would in fact be more random than `nanotime()` even on Linux/MacOS/Windows/WASI.

Reply:
Switched to hardware randomess in https://github.com/tinygo-org/tinygo/pull/4476/commits/a47c0c1386eefbf642c8c7c27e79d9e29ef03a71